src/Extension/FluentExtension.php

```diff
@@ -638,21 +638,27 @@ public function onAfterWrite(): void
     public function onAfterDuplicate($original, $doWrite, $relations): void
     {
         $owner = $this->getOwner();
-        $localisedTables = $this->owner->getLocalisedTables();
+        $localisedTables = $owner->getLocalisedTables();
+
         // Get the names of all has_one columns that will have new IDs
         $copyRelations = (array) $owner->config()->get('localised_copy');
+        $cascadeDuplicates = (array) $owner->config()->get('cascade_duplicates');
         $hasOne = $owner->hasOne();
         $copyHasOneRelations = [];
+
         foreach ($copyRelations as $relationName) {
-            if (array_key_exists($relationName, $hasOne)) {
-                $copyHasOneRelations[] = $relationName . 'ID';
+            // We only need to process has_one relations which are covered in both localised_copy & cascade_duplicates
+            // because only such cases need special handling for duplication
+            if (array_key_exists($relationName, $hasOne) && in_array($relationName, $cascadeDuplicates)) {
```

Comment:
```suggestion
            // We only need to process has_one relations which are covered in localised_copy and are part of this duplication
            // because only such cases need special handling for duplication
            if (array_key_exists($relationName, $hasOne) && in_array($relationName, $relations)) {
```
The `$relations` array contains the relations which are relevant for this duplication action. Note that you can pass in a custom array to `duplicate()` which will be used instead of `cascade_duplicates`.

Please also add a test scenario where a custom `$relations` array is passed into `duplicate()` which differs from the `cascade_duplicates` configuration array.